From cca12cb32f3c8338476e9a6370ed15760ffb7bf5 Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Mon, 26 Nov 2018 18:29:30 +0200
Subject: [PATCH 352/706] dts: ls1028a: rdb: Add w/a for mdio h/w issue by
 using fixed link phys on RDB

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit b2354e9bb9566945d9becc34d4f25fca3c0e9ef7)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../boot/dts/freescale/fsl-ls1028a-rdb.dts    | 39 +++++++++++++------
 1 file changed, 27 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
index 93ff9282211a..9d50d5a90189 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
@@ -255,9 +255,12 @@
 	};
 };
 
-&enetc_port0 {
-	phy-handle = <&sgmii_phy0>;
-	phy-connection-type = "sgmii";
+&enetc_port0 { /* workaround MDIO issue, normally it's sgmii_phy0 */
+	phy-connection-type = "internal";
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
 };
 
 &enetc_port1 {
@@ -281,24 +284,36 @@
 };
 
 /* l2switch ports */
-&switch_port0 {
-	phy-handle = <&qsgmii_phy1>;
-	phy-connection-type = "qsgmii";
+&switch_port0 { /* workaround MDIO issue, normally it's gsgmii_phyX */
+	phy-connection-type = "internal";
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
 };
 
 &switch_port1 {
-	phy-handle = <&qsgmii_phy2>;
-	phy-connection-type = "qsgmii";
+	phy-connection-type = "internal";
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
 };
 
 &switch_port2 {
-	phy-handle = <&qsgmii_phy3>;
-	phy-connection-type = "qsgmii";
+	phy-connection-type = "internal";
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
 };
 
 &switch_port3 {
-	phy-handle = <&qsgmii_phy3>;
-	phy-connection-type = "qsgmii";
+	phy-connection-type = "internal";
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
 };
 
 &edp {
-- 
2.17.1

